R-package/R/lgb.Dataset.R

```diff
@@ -186,7 +186,7 @@ Dataset <- R6::R6Class(
       if (!is.null(private$reference)) {
         ref_handle <- private$reference$.__enclos_env__$private$get_handle()
       }
-      handle <- NA_real_
+      handle <- 0L
```

Comment:
maybe there is a `NA_integer_` ?

Reply:
ha yes I was just typing that! These should be `NA_integer_` 

https://stat.ethz.ch/R-manual/R-devel/library/base/html/NA.html